### DISCUSSION TITLE
It is incredibly hard to contribute to the SDK due to vastly outdated framework usages and not automated build verification

### DISCUSSION THREAD

**Discussion by danielmarbach** [aws/aws-sdk-net#2559] [General]
Normally, I would always send in contributions to something. In fact have contributed over 80 pull requests to the Azure .NET SDK to improve things and smoothen out rough edges. Unfortunately, the AWS .NET SDK uses:

- A very outdated NET 45 dependency
- Has no automated build pipeline that execute my set of changes against a verification suit
- Is tied to Windows which makes it incredibly hard to even run tests locally without going through significant hurdles

I personally primarily develop these days on Linux. 

The entry barrier to contributing to the SDK is too high for me at the moment, which is sad.

// cc @normj

**Comment by normj** [comment]
I agree it is hard to contribute to the AWS .NET SDK. A lot of the reason is the wide range of supported platform the SDK, a repository structure that was first setup when AWS was a lot smaller and when .NET was Windows only.

We do have ambition to make things better. Right now we are in a big rework of our internal build system used to meet the demands of the daily release cycle of the SDK. After that hopefully we can start making the developer cycle for contributors as well as for the AWS .NET team better. We have to make changes very careful because every day the SDK has to be ready to ship with no surprises. The analogy of trying to change a car's oil while driving down the freeway is how structural changes to the SDK repository can be looked at.
